internal/test/providers/cloudprovider.go

```diff
@@ -0,0 +1,33 @@
+package providers
+
+import (
+	"fmt"
+
+	"github.com/openshift/api/config/v1"
+	mapi "github.com/openshift/machine-api-operator/pkg/apis/machine/v1beta1"
+	oc "github.com/openshift/windows-machine-config-bootstrapper/internal/test/clusterinfo"
+	awsProvider "github.com/openshift/windows-machine-config-bootstrapper/internal/test/providers/aws"
+	"github.com/pkg/errors"
+)
+
+type CloudProvider interface {
+	GenerateMachineSet(bool, int32) (*mapi.MachineSet, error)
+}
+
+func NewCloudProvider(sshKeyPair string) (CloudProvider, error) {
```

Comment:
You typically want to add methods in the commit where they get used. Yes, it might make the commit large but it implies that how you organize the commits needs to change. But given this is for e2e testing, I won't block on this change.

Reply:
All of this looks like boilerplate code that also exists in the operator repo. There is a lot of duplication going on here.

Reply:
We decided to copy the code in design meet as we dont want to import code from operator repo in WMCB test. That explains a different commit.